x-pack/plugins/security_solution/public/timelines/components/row_renderer_switch/index.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiSwitchEvent } from '@elastic/eui';
+import { EuiToolTip, EuiSwitch, EuiFormRow, useGeneratedHtmlId } from '@elastic/eui';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { RowRendererId } from '../../../../common/api/timeline';
+import type { State } from '../../../common/store';
+import { setExcludedRowRendererIds } from '../../store/actions';
+import { selectExcludedRowRendererIds } from '../../store/selectors';
+import * as i18n from './translations';
+
+interface RowRendererSwitchProps {
+  timelineId: string;
+}
+
+const CustomFormRow = styled(EuiFormRow)`
+  .euiFormRow__label {
+    font-weight: 400;
+  }
+`;
+
+export const RowRendererSwitch = React.memo(function RowRendererSwitch(
+  props: RowRendererSwitchProps
+) {
+  const toggleTextSwitchId = useGeneratedHtmlId({ prefix: 'rowRendererSwitch' });
+
+  const { timelineId } = props;
+
+  const dispatch = useDispatch();
+
+  const excludedRowRendererIds = useSelector((state: State) =>
+    selectExcludedRowRendererIds(state, timelineId)
+  );
+
+  const [checked, setChecked] = useState(
+    !Object.values(RowRendererId).every((id) => excludedRowRendererIds.includes(id))
+  );
+
+  const updateExcludedRowRenderers = useCallback(
+    (payload) =>
+      dispatch(
+        setExcludedRowRendererIds({
+          id: timelineId,
+          excludedRowRendererIds: payload,
+        })
+      ),
+    [dispatch, timelineId]
+  );
+
+  const handleDisableAll = useCallback(() => {
+    updateExcludedRowRenderers(Object.values(RowRendererId));
+  }, [updateExcludedRowRenderers]);
+
+  const handleEnableAll = useCallback(() => {
+    updateExcludedRowRenderers([]);
+  }, [updateExcludedRowRenderers]);
```

Comment:
same here actually, we have these 2 use callbacks that are used only once each in the `onChange` method below. So now we have a `useCallback` used in these 2 `useCallback` used in the `onChange` below...
I feel like using the dispatch functions directly in the `onChange` below would be more readable

Reply:
I have no preference regardless. Fixed here : a47d2803e3edf440089cb94144b4a32352ff9fe1